backtracking/power_sum.py

```diff
@@ -0,0 +1,83 @@
+"""
+Problem source: https://www.hackerrank.com/challenges/the-power-sum/problem
+Find the number of ways that a given integer X, can be expressed as the sum
+of the Nth powers of unique, natural numbers. For example, if X=13 and N=2.
+We have to find all combinations of unique squares adding up to 13.
+The only solution is 2^2+3^2. Constraints: 1<=X<=1000, 2<=N<=10.
+"""
+
+from math import pow
+
+sum_ = count_ = 0
+
```

Comment:
the variables `sum_` and `count_` are not very descriptive. It would be better to use more descriptive names like `current_sum` and `solution_count`, or anything which makes it more descriptive.

Reply:
Fixed